### PR TITLE
Rust: generate the extractor

### PR DESCRIPTION
**Pull request by aibaars**
This pull request splits the `translate.rs` file of the extractor in two parts, the non-generated part (`base.rs`) and the generated code `generated.rs`. 

The extractor can be regenerated by running `cargo run` in `rust/generate-schema` .

### CODE REVIEW

**Comment by Safay99** [comment]
- [ ]

**Comment by Safay99** [comment]
aibaars/rust-gen-extractor

**Comment by Safay99** [comment]
1. aibaars/rust-gen-extractor

File: rust/extractor/src/translate/generated.rs [hunk 1]
```diff
@@ -0,0 +1,2185 @@
+use super::base::{TextValue, Translator};
```
Comment (hvitved) [inline rust/extractor/src/translate/generated.rs:1]: I think it's nice when generated code starts with a comment like `generated code, do not edit`.

File: rust/extractor/src/main.rs [hunk 2]
```diff
@@ -1,11 +1,43 @@
...
+use ra_ap_parser::Edition;
 mod archive;
 mod config;
 pub mod generated;
 mod translate;
 pub mod trap;
+use ra_ap_syntax::ast::SourceFile;
+use ra_ap_syntax::AstNode;
 
+pub fn extract(
```
Comment (hvitved) [inline rust/extractor/src/main.rs:12]: Does this need to be public?

File: rust/extractor/src/translate/generated.rs (hunk 1 above)
Comment (aibaars) [reply rust/extractor/src/translate/generated.rs:3]: Yes, good idea; I also added some `linguist-generated` entries to `.gitattributes`

File: rust/extractor/src/main.rs (hunk 2 above)
Comment (aibaars) [reply rust/extractor/src/main.rs]: No, it can be private.

**Review by hvitved — APPROVED** [review]

**Pull request merged**